core/lib/dal/src/blocks_dal.rs

```diff
@@ -2294,6 +2369,19 @@ impl BlocksDal<'_, '_> {
         self.map_storage_l1_batch(l1_batch).await
     }
 
+    pub async fn get_l1_batch_metadata_only(
+        &mut self,
+        number: L1BatchNumber,
+    ) -> DalResult<Option<L1BatchMetadata>> {
+        let Some(storage_batch) = self.get_storage_l1_batch(number).await? else {
+            return Ok(None);
+        };
+        let Ok(metadata) = storage_batch.clone().try_into() else {
+            return Ok(None);
+        };
```

Comment:
why not an error?

Reply:
It follows the same logic as in `get_l1_batch_metadata`. Erroring should also be fine, probably better.

Reply:
I tried making this into error, but then I would need to change the function signature. Currently it returns `DalResult` that implies having `DalError`. There does not seem to be a good way to map this conversion error into it.